### PR TITLE
Parallel Gelu with ParallelFor

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/activations.h [hunk 1]
```diff
@@ -38,6 +39,34 @@ class Gelu : public OpKernel {
   Status Compute(OpKernelContext* context) const override {
     const auto* X = context->Input<Tensor>(0);
     Tensor* Y = context->Output(0, X->Shape());
+    concurrency::ThreadPool* tp = context->GetOperatorThreadPool();
+    if (nullptr != tp) {
```
Comment (snnn) [inline onnxruntime/contrib_ops/cpu/activations.h]: else !!!!

File: onnxruntime/contrib_ops/cpu/activations.h [hunk 2]
```diff
@@ -38,6 +39,34 @@ class Gelu : public OpKernel {
...
+      const T* input = X->template Data<T>();
+      T* output = Y->template MutableData<T>();
+      int thread_num = tp->NumThreads();
```
Comment (snnn) [inline onnxruntime/contrib_ops/cpu/activations.h]: You may plus it to one.

Comment (skottmckay) [reply onnxruntime/contrib_ops/cpu/activations.h]: > You may plus it to one.

ugh. we really need to fix this. a user (internal or external) shouldn't need to know such things. NumThreads should return the number of threads the threadpool will use.

Comment (snnn) [reply onnxruntime/contrib_ops/cpu/activations.h]: Yes, we should fix this. But NumThreads() doesn't know if the caller is in the threadpool or not.

Comment (snnn) [reply onnxruntime/contrib_ops/cpu/activations.h]: The hole was digger by @yufenglee himself

Comment (yufenglee) [reply onnxruntime/contrib_ops/cpu/activations.h]: @snnn the real matter is whether we should use the main thread for calculation or not in ParallelFor. I remembered you've ever made a change to not use main thread for calculation and triggered a bad performance regression, and you reverted your change, i.e., doing calculation in main thread get better performance.

Comment (snnn) [reply onnxruntime/contrib_ops/cpu/activations.h]: Correct.

**Pull request merged**